src/scipp/experimental/plotting/wrappers.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from typing import Union, Dict
+from .model import input_node
+from .figure import Figure
+from ... import Variable, DataArray, Dataset, arange
+from ...typing import VariableLike
+import numpy as np
+
+
+def _to_data_array(obj):
+    out = obj
+    if isinstance(out, np.ndarray):
+        dims = [f"axis-{i}" for i in range(len(out.shape))]
+        out = Variable(dims=dims, values=out)
+    if isinstance(out, Variable):
+        out = DataArray(
+            data=out,
+            coords={dim: arange(dim, size)
+                    for dim, size in out.sizes.items()})
```

Comment:
Minor: Could omit setting coords here because they are added in the loop below anyway.